plugins/coverimage.koplugin/main.lua

```diff
@@ -158,16 +197,440 @@ function CoverImage:onReaderReady(doc_settings)
+        or (cache_size_KiB > self.cover_image_cache_maxsize * 1024 and self.cover_image_cache_maxsize ~= 0)
+        and index <= #files do
+        os.remove(files[index].name)
+        cache_count = cache_count - 1
+        cache_size_KiB = cache_size_KiB - files[index].size
+        index = index + 1
+    end
+    logger.dbg("CoverImage: clean - cache size: ".. cache_size_KiB .. " KiB, cached files: " .. cache_count)
+end
+
+function CoverImage:isCacheEnabled(path)
+    if not path then
+        path = self.cover_image_cache_path
+    end
+    if Device.isAndroid() then
+        return self.cover_image_cache_maxfiles >= 0 and self.cover_image_cache_maxsize >= 0
+            and lfs.attributes(path, "mode") == "directory"
+            and path ~= "/sdcard/koreader/cache/"  -- interferes with frontent cache-framework
+            and realpath(path) ~= realpath(android.getExternalStoragePath() .. "/koreader/cache/")  -- interferes with frontent cache-framework
+    else
+        return self.cover_image_cache_maxfiles >= 0 and self.cover_image_cache_maxsize >= 0
+            and lfs.attributes(path, "mode") == "directory"
+            and path ~= "./cache/" -- interferes with frontent cache-framework; quick and dirty check
+            and realpath(path) ~= realpath("./cache/") -- interferes with frontent cache-framework; quick and dirty check
+    end
+end
+
+-- callback for choosePathFile()
+function CoverImage:migrateCache(old_path, new_path)
+    if old_path == new_path or not self:isCacheEnabled(new_path) then
+        return
+    end
+    for entry in lfs.dir(old_path) do
+        if entry ~= "." and entry ~= ".."   then
+            local old_file = old_path .. entry
+            if lfs.attributes(old_file, "mode") == "file" and entry:sub(1,self.cover_image_cache_prefix:len()) == self.cover_image_cache_prefix then
+                local old_access_time = lfs.attributes(old_file, "access")
+                local new_file = new_path .. entry
+                os.rename(old_file, new_file)
+                lfs.touch(new_file, old_access_time) -- restore original time
+            end
+        end
+    end
+end
+
+-- callback for choosePathFile()
+function CoverImage:migrateCover(old_file, new_file)
+    if old_file ~= new_file then
+        os.rename(old_file, new_file)
+    end
+end
+
+--[[--
+chooses a path or (an existing) file
+
+@touchmenu_instance for updating of the menu
+@string setting is the G_reader_setting which is used and changed
+@boolean folder_only just selects a path, no file handling
+@boolean new_file allows to enter a new filename, or use just an existing file
+@function migrate(a,b) callback to a function to mangle old folder/file with new folder/file.
+    Can used for migrating the contents of the old path to the new one
+]]
+function CoverImage:choosePathFile(touchmenu_instance, setting, folder_only, new_file, migrate)
+    local old_path, old_name = util.splitFilePathName(self[setting]) -- luacheck: no unused
```

Comment:
We have `__` and `dummy` placeholders whitelisted for that when `_` is taken by gettext ;).